utilities/infra.py

```diff
@@ -881,35 +884,48 @@ def verify_cluster_sanity(
         junitxml_property (property): Junitxml property
 
     """
-    skip_cluster_sanity_check = "--cluster-sanity-skip-check"
-    skip_rhoai_check = "--cluster-sanity-skip-rhoai-check"
+    # Option to completely skip the check (can co-exist)
+    skip_check_opt = "--cluster-sanity-skip-check"
+    # Option to skip specific RHOAI parts
+    skip_rhoai_check_opt = "--cluster-sanity-skip-rhoai-check"
+    # Option to control behavior *on failure*
+    continue_on_failure_opt = "--cluster-sanity-continue-on-failure"
 
-    if request.session.config.getoption(skip_cluster_sanity_check):
-        LOGGER.warning(f"Skipping cluster sanity check, got {skip_cluster_sanity_check}")
-        return
+    # Handle the option to completely skip the check first
+    if request.session.config.getoption(skip_check_opt):
+        LOGGER.warning(f"Skipping cluster sanity check entirely due to {skip_check_opt} flag.")
+        return  # Exit the function early, tests will run without checks
 
     try:
-        LOGGER.info("Check cluster sanity.")
-
+        LOGGER.info("Running cluster sanity check...")
```

Comment:
I think we should not skip here. If the nodes are not schedulable or healthy. It is a cluster problem.

Reply:
This is different from what was being discussed in slack. I do agree with you, but can we get an ack to proceed before implementing the change?